src/nlp.jl

```diff
@@ -328,7 +429,25 @@ function MathProgBase.eval_jac_g(d::JuMPNLPEvaluator, J, x)
             idx += 2
         end
     end
-    eval_jac_g!(subarr(J,idx:length(J)), (d.m.nlpdata::NLPData).nlconstrlist, x)
+    grad_storage = d.jac_storage
+    subexpr_reverse_values = d.subexpression_reverse_values
+    for ex in d.constraints
+        nzidx = ex.grad_sparsity
+        grad_storage[nzidx] = 0.0
+        subexpr_reverse_values[ex.dependent_subexpressions] = 0.0
+
+        reverse_eval(grad_storage,ex.reverse_storage,ex.forward_storage,ex.nd,ex.adj,subexpr_reverse_values,1.0)
+        for i in length(ex.dependent_subexpressions):-1:1
+            k = ex.dependent_subexpressions[i]
+            subexpr = d.subexpressions[k]
+            reverse_eval(grad_storage,subexpr.reverse_storage,subexpr.forward_storage,subexpr.nd,subexpr.adj,subexpr_reverse_values,subexpr_reverse_values[k])
+        end
+
+        for k in 1:length(nzidx)
+            J[idx+k-1] = grad_storage[nzidx[k]]
+        end
+        idx += length(nzidx)
+    end
 
     d.eval_jac_g_timer += toq()
     #print("x = ");show(x);println()
```

Comment:
Same - is this intentionally left commented?

Also raises the idea of adding in a proper logging library so that you could make something like this a debug statement